database/migrations/2021_07_04_153303_add_prenotes_columns_to_controller_positions_table.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class AddPrenotesColumnsToControllerPositionsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('controller_positions', function (Blueprint $table) {
+            $table->boolean('sends_prenotes')
+                ->after('receives_departure_releases')
+                ->default(false)
+                ->comment('Whether the controller position can send prenotes');
+
+            $table->boolean('receives_prenotes')
+                ->after('sends_prenotes')
+                ->default(false)
+                ->comment('Whether the controller position can receive prenotes');
+        });
+
+        DB::table('controller_positions')
+            ->update(
+                [
+                    'sends_prenotes' => DB::raw('`requests_departure_releases`'),
```

Comment:
What is this needed for?

Reply:
So the database fields the controller-positions dependency in the plugin. It's essentially to improve the UX plugin-side. It prevents positions that can't receive prenotes from appearing in the controller list / prevents people who can't request from requesting.

In theory, if someone can approve a release, they can probably accept a prenote, so this is a good starting point :)